src/vs/workbench/contrib/positronPlots/browser/components/plotsContainer.tsx

```diff
@@ -82,7 +87,49 @@ export const PlotsContainer = (props: PlotContainerProps) => {
 				plotHistory.scrollTop = plotHistory.scrollHeight;
 			}
 		}
-	});
+	}, [plotHistoryRef]);
+
+	useEffect(() => {
+		// Be defensive against null sizes when pane is invisible
+		if (plotWidth <= 0 || plotHeight <= 0) {
+			return;
+		}
+
+		const notify = () => {
+			let policy = props.positronPlotsService.selectedSizingPolicy;
+
+			if (policy instanceof PlotSizingPolicyIntrinsic) {
+				policy = new PlotSizingPolicyAuto;
+			}
+
+			const viewPortSize = {
+				height: plotHeight,
+				width: plotWidth,
+			}
+			let size = policy.getPlotSize(viewPortSize);
+			size = size ? size : viewPortSize;
+
+			props.positronPlotsService.setPlotsRenderSettings({
+				size,
+				pixel_ratio: DOM.getActiveWindow().devicePixelRatio,
+				format: PlotRenderFormat.Png, // Currently hard-coded
+			});
+		};
+
+		// Renotify if the sizing policy changes
+		const disposables = new DisposableStore();
+		disposables.add(props.positronPlotsService.onDidChangeSizingPolicy((_policy) => {
+			notify();
+		}));
+
+		// Propagate current render settings. Use a debouncer to avoid excessive
+		// messaging to language kernels.
+		const debounceTimer = setTimeout(() => {
+			notify()
+		}, 500);
+
+		return () => clearTimeout(debounceTimer);
```

Comment:
The function returned from `useEffect` needs to dispose of the `disposables` as well

Reply:
oops!